### PR TITLE
Add Vision doc to governance repo; minor edits

### PR DESCRIPTION
**Pull request by CharliePoole**
Start of #10 

Looking at the doc on the wiki, I saw it was still worded as things we __were planning to do__ in NUnit 3.0. I modified the wording and corrected a few errors.

If you wait an hour or so before reviewing this, I'll add annotations about where I would like to go with it.

### CODE REVIEW

File: vision.md
Comment (CharliePoole) [inline vision.md:4]: This sentence is the actual vision statement. The rest is explanation.

In NUnit 2 it said simply "NUnit is a unit testing framework for developers working with the .NET framework."

IMO this is the most important thing on the page. We need to word it to say what NUnit is going to be.

I would leave it as is or possibly get rid of the notion of a "platform" here and later, since we really don't emphasize it any more.

File: vision.md
Comment (CharliePoole) [inline vision.md:7]: I would remove this note. It's old stuff now.

File: vision.md
Comment (CharliePoole) [inline vision.md:26]: In the three preceding paragraphs, I don't think we need the comparison with NUnit 2. That applies to any more that comes up below, so I won't repeat this comment.

File: vision.md
Comment (CharliePoole) [inline vision.md:30]: Do we want to have the "NUnit as a Platform" stuff? I'm doubtful here.

File: vision.md
Comment (CharliePoole) [inline vision.md:32]: I don't think the future directions section is needed at all. But if there are parts of it you want me to keep, please let me know.

File: vision.md
Comment (CharliePoole) [inline vision.md:42]: I think that this is the first of a set of what I would call "Principles" that we enunciated and ought to be following - unless we decide to drop some of them. My inclination is to list them that way - as principles we try to follow.

File: vision.md
Comment (CharliePoole) [inline vision.md:43]: Principle 2 - I would drop some of the specifics which are outdated.

File: vision.md
Comment (CharliePoole) [inline vision.md:46]: Principle 3

File: vision.md
Comment (CharliePoole) [inline vision.md:48]: Principle 4

File: vision.md
Comment (CharliePoole) [inline vision.md:49]: Principle 5 - except we haven't done it so far. Keep or drop?

File: vision.md
Comment (CharliePoole) [inline vision.md:51]: Principle 6

Any others?

**Comment by CharliePoole** [comment]
I added comments to help in reviewing. I don't plan to change this before we merge because it now matches the published wiki NUnit Vision page. We should merge it and then use it as the basis for further changes.

File: vision.md
Comment (rprouse) [reply vision.md:26]: Agreed, we should be looking forward and stating our desired direction, not looking back or going too much into the history.

File: vision.md
Comment (rprouse) [reply vision.md:4]: I think this should be our one missions statement, an elevator pitch for NUnit that sums up everything below. I think the current statement is too generic and doesn't actually say anything. I am going to propose some words to get it started, but feel free to rework and remix,

> NUnit aims to be a best in breed test framework for .NET that enables end to end testing of .NET projects by being unopinionated and allowing many testing styles.

I like the `unopinionated` because other test frameworks are often regarded as opinionated, ie forcing you to write tests one way and I think we should embrace being open. That is why I use NUnit and joined the team in the first place 😄 

We might also want to add something about supporting all commonly used .NET platforms and maintaining support for older platforms.

File: vision.md
Comment (rprouse) [reply vision.md:30]: No, I don't like the concept. I say remove it here and above.

File: vision.md
Comment (rprouse) [inline vision.md:35]: Actually, I kind of like this stuff except the `Taking the above...` paragraph above. I think we should keep as principles from this para on.

File: vision.md
Comment (rprouse) [reply vision.md:42]: Remove J# and I would remove Iron Python/Ruby too. .NET Languages change and go in and out of favor, so maybe not even list the ones we support, just stick with the first sentence?

File: vision.md
Comment (rprouse) [reply vision.md:49]: This is a tough one. I think it is a worthy goal, but knowing the number of strings we have and how they are used, I doubt we will ever get to adding in resource DLLs. It also feels opposite to what our recent stance has been which is striving for culture neutral.

We also have code here and there that reads our error messages, let alone the many unit tests that compare messages. Does anyone think we will ever get the work done?

If someone wants to champion it however, I will support them.

**Review by rprouse — COMMENTED** [review]
Should we have a principle on how long we support legacy versions of the .NET Framework? For example, at some point we might need to only release a .NET 4+ console runner or drop support for .NET 2.0. Should we link our support to Microsoft's LTS for versions, or longer? Should this be a separate issue for the team to decide?

I am not pushing to drop anything here, but at some point in the future we will need to make these decisions just like the decision was made for .NET 1.x.

File: vision.md
Comment (jnm2) [reply vision.md:42]: C++/CLI will be next to disappear, I think.

File: vision.md
Comment (jnm2) [reply vision.md:46]: Instead of MS.NET I'd use the full intended names of the separate products; **.NET Framework** and **.NET Core**. What should we do with the line saying that we will expand to support Compact Framework and Silverlight?

File: vision.md
Comment (jnm2) [reply vision.md:48]: s/programatically/programmatically

File: vision.md
Comment (jnm2) [reply vision.md:49]: I'd like to keep this as an attainable goal and wait until people start asking for it to prioritize it.

**Comment by jnm2** [comment]
> Should we have a principle on how long we support legacy versions of the .NET Framework?

Sounds good.

**Comment by CharliePoole** [comment]
Guys, a question about how you want to do this procedurally.

I had asked to just copy over the existing Vision doc and __then__ modify it, get comments, re-modify, etc.

Some of you (I think Rob and Chris) objected and asked for a PR just to do the move. I said OK, I'll do two PRs, one for the move one for editing. 

I'm getting a lot of comments here that feel like they are intended to define the final version. That's OK but I do not think they should be implemented in this PR because you will then be changing what was already changed when we do the final version. You will loose sight of the base that is being modified and the changes that may need to be explained to users. Remember the existing Vision has been the bible of NUnit since around 2008.

Nevertheless, I'm happy (well willing) to do it in any way you want, but I'd like us to decide before I add any comments onto your comments. 

PS I don't consider this a decision requiring a formal vote. I'll see what the consensus is and act on it. Copying a file into a folder != voting on a governance policy.

File: vision.md
Comment (CharliePoole) [reply vision.md:42]: I agree with one provision that I will to include in some text you can comment on in the next PR. Languages "disappear" from the microsoft point of view but not necessarily from the wild. We should support what users use not what microsoft supports.

Comment (CharliePoole) [reply vision.md:42]: BTW, I know I said I'd wait for the answer to my general comment before making line comments, but I couldn't resist.

File: vision.md
Comment (CharliePoole) [reply vision.md:46]: I didn't repeat the comment from above, but I will remove out of date stuff in the actual edit.

File: vision.md
Comment (CharliePoole) [reply vision.md:49]: Maybe we want "internationalizable" rather than "internationalized". We do the former, support others in doing the latter.

**Comment by rprouse** [comment]
I thought we were going to just do a PR and merge as-is, then do edits, but was confused by your comments on how you wanted to change the document, so thought you wanted to edit it now. I am good either way.

**Comment by jnm2** [comment]
\<humor>
[Okay, I'm with you fellers. 😃](https://youtu.be/wYXFSbLotSU)
\</humor>

**Comment by CharliePoole** [comment]
@rprouse 
I don't know what "do a PR and merge as is" means. This PR, which says "Start of #10" contains no changes from what's already in the wiki. It is the first of two PRs. The second one was to have the changes.

[This two-PR process seems to me to be a bit silly, but it's what this committee - or two members anyway - asked for. FYI committees I have seen usually work that way... if some members want a strict process followed but others do not care, the stricter process is usually followed. That's what happened here. Thank goodness we don't have to work this way on our actual software!]

Anyway, this PR is to merge the text of an existing document from our wiki into the governance repo. It is unchanged except for a few refs to being on the wiki and the title. It needs to be approved or rejected. If rejected for reasons of process, I need to know what you want to do instead.

@jnm2 Yes, funny. Just not that funny, to me, right now. 😞

**Comment by jnm2** [comment]
@CharliePoole I'm sorry. 😞 I'm not sure everyone was understanding each other in #10. I agree, acting on this PR's line comments in a second PR seems unwieldy. I don't care whether there are one or two PRs so long as no changes in the substance of the text are squashed into the commit initially moving the document over, so that the diff exists in this repo.

Since @rprouse and @ChrisMaddock are the ones that communicated to you that they wanted two PRs, I'll let them review first because they have a clearer idea what they're looking for in this PR.

**Comment by rprouse** [comment]
@CharliePoole I stated above that I thought we were doing two PRs to do a straight merge with the first. I only wanted to do two PR's because master is locked and it is best if we don't push directly to master. I only added comments to this PR because you did first so I thought you wanted a response. To be honest, **I don't really give a fuck** and I am getting sick of this. NUnit has been running fine for several years now without all this process and ceremony. You feel it is necessary which is fine, but let's just do it.

I am going to back off for awhile. I am not interested in this petty bullshit and as you can tell it is pissing me off. I am only trying to be helpful, but you see to be misinterpreting everything I say. Issues and PRs are not the best way to communicate and context is often lost. That is clearly happening and I am sorry if I have not been making my intentions clear enough.

**Comment by CharliePoole** [comment]
@rprouse 
You didn't give any reason for wanting two PRs - i.e. because master is locked. I took the reason to be a lack of trust. I apologize for the misunderstanding. In any case, I didn't complain at the time, but when you didn't approve it for merge and everyone else followed suit, I got annoyed and my tone no doubt showed it. However, unless somebody else does something, this PR can't move forward.

I'm not in love with all the process and ceremony either. I think you need to look at the writing process. I  asked people some questions and wrote drafts. The drafts reflected what I was told - or at least what I thought I was told. I tried not to inject my own views. We had reviews. People told me to add or change stuff. Almost nobody told me to take anything out. Committee writing seems to work that way. If it's too much, someone could go through and simplify it.

**Review by jnm2 — APPROVED** [review]
Looks good for initial move.

**Comment by rprouse** [comment]
Since I started this in public, I will try to close it off in public. I think we are all trying to do the right thing, but nerves are getting a bit frayed by the process. I am sorry for anything I have done to contribute to that. I think I need to say my piece though...

@CharliePoole it is not just this PR that is upsetting me, this is just the latest example. It seems like ever since @OsirisTerje and I changed our minds about the adapter release last fall, you have become frustrated with people not sticking with plans and it feels like whenever anyone does that, you respond in a very passive aggressive way. To me, you have been taking minor misunderstandings and expanding them into unneeded back and forth.

Another recent example is the debate over dropping or obsoleting `AssetHelper`. I know you did work separating it out and I am sorry for that, but the amount of discussion it raised was more than the work involved. I don't care whether we drop it or move it, I just don't think we can do either in one release, we need to inform users and since 99% of developers who use NUnit probably don't even read the release notes, the best way to do so is to put it in their face by Obsoleting something even if it doesn't meet a strict definition of Obsolete.

As you stated in a separate email, I think the transition process has gone on two long and the resulting ambiguity has led the two of us to step on each others' toes causing these problems.

Let's push through this and get it done. I am fine with this vision document as is. I am also fine with the core team addressing any changes at a later date. We've been trying to get everything in place, but that isn't necessary right now. Let's just get done what needs to get done and move forward.

File: vision.md
Comment (ChrisMaddock) [reply vision.md:35]: I agree.

**Review by rprouse — APPROVED** [review]

File: vision.md
Comment (ChrisMaddock) [reply vision.md:49]: I think this is too low-level. Should we one day encounter a German speaker, who wishes to contribute and maintain a translation, we deal with that as it comes up. Having this here implies it's a goal we're working towards, and I don't think that should be a high-priority.

**Review by ChrisMaddock — APPROVED** [review]
I'm confused as to how all this has arisen, as far as I can see, myself and Rob only agreed with what you proposed on #10, @CharliePoole.

Anyway, it doesn't matter. At this point, as the one proposing the changes, I suggest you proceed as you see fit Charlie, and let us know at which point you want reviews on the final document. I've commented on the high-level issues only, below.

**Comment by ChrisMaddock** [comment]
(Apologies Rob, our comments overlapped. I don't think I contradicted you too heavily!)

File: vision.md
Comment (CharliePoole) [reply vision.md:35]: Will keep.

File: vision.md
Comment (CharliePoole) [reply vision.md:49]: @ChrisMaddock  I don't think it's a matter of level. If a project makes it a top goal, then it's a top goal. The project goals I defined in 2008 had this as a deliberate top-level goal. In fact we never did it and it sounds like nobody wants it as a goal right now. I'd say we drop it because it's not our goal any more. IMO, you have to be ruthless about not keeping as goals things you "might" want to do some day.

**Pull request merged**
